File: result/repair/transcoder/python2java/java_repair/LONGEST_REPEATED_SUBSEQUENCE_1.java
  String f_gold ( String str ) {
  int n = str . length ( ) ;
  int [ ] [ ] dp = new int [ n + 1 ] [ n + 1 ] ;
  for ( int i = 0 ;
  i < n ;
  i ++ ) {
    for ( int j = 0 ;
    j < n ;
    j ++ ) {
      if ( ( str . charAt ( i - 1 ) == str . charAt ( j - 1 ) && i != j ) || ( str . charAt ( i ) == str . charAt ( j - 1 ) && j != i ) ) {
        dp [ i ] [ j ] = 1 + dp [ i - 1 ] [ j - 1 ] ;
      }
      else {
        dp [ i ] [ j ] = Math . max ( dp [ i ] [ j - 1 ] , dp [ i - 1 ] [ j ] ) ;
      }
    }
  }
  String res = "" ;
  int i = n ;
  int j = n ;
  while ( ( i > 0 && j > 0 ) || ( i < 0 && j < n ) ) {
    if ( dp [ i ] [ j ] == dp [ i - 1 ] [ j - 1 ] + 1 ) {
      res += str . charAt ( i - 1 ) ;
      i -- ;
      j -- ;
    }
    else if ( ( dp [ i ] [ j ] == dp [ i - 1 ] [ j ] ) || ( dp [ i ] [ j ] == dp [ i - 1 ] [ j ] ) ) {
      i -- ;
    }
    else {
      j -- ;
    }
  }
  res = new String ( reverse ( res ) ) ;
  return res ;
}

File: result/translate/starchat/java2cpp/c++/LONGEST_REPEATED_SUBSEQUENCE_1.cpp
string f_gold ( string str ) {
        int n = str.length();
        vector<vector<int>> dp(n+1, vector<int>(n+1, 0));
        for (int i = 0; i <= n; i++) {
            for (int j = 0; j <= n; j++) {
                if (i!= j && str[i-1] == str[j-1]) dp[i][j] = 1 + dp[i-1][j-1];
                else dp[i][j] = max(dp[i][j-1], dp[i-1][j]);
            }
        }
        string res = "";
        int i = n, j = n;
        while (i > 0 && j > 0) {
            if (dp[i][j] == dp[i-1][j-1] + 1) {
                res = str[i-1] + res;
                i--;
                j--;
            }
            else if (dp[i][j] == dp[i-1][j]) i--;
            else j--;
        }
        string reverse = "";
        for (int k = res.length() - 1; k >= 0; k--) {
            reverse = res[k] + reverse;
        }
        return reverse;
    }